drivers/docker_driver.go

```diff
@@ -144,9 +144,12 @@ func (d *DockerDriver) retrieveTar(t *testing.T, path string) (*tar.Reader, erro
 		Path:         path,
 	})
 	if err != nil {
+		// attempt to clean up container before returning
+		d.removeContainer(t, container.ID)
```

Comment:
Would it be easier to throw this into a defer?

Reply:
yep thats a bit more clear, updated